### ISSUE TITLE
Static peers can get removed by autopeering

### ISSUE THREAD

**Issue by WernerderChamp** [iotaledger/hornet#474]
**Describe the bug**
Under certain conditions, static peers can get removed during runtime by autopeering. They will no longer appear in the neighbor list and the affected peer will receive "connection refused" if trying to connect to the node. Restarting hornet will bring the peer back (I have not tried reloading the peering.json, might be worth a try). 

**To Reproduce**
 -Get a static peer
 -Wait for it to eventually get disappear from your neighbor list.

**Expected behavior**
Static peers should stay connected, completely unaffected of autopeering

**Environment information:**
 - OS: 20.04
 - RAM: 8 GB
 - Cores: 4 Cores
 - Type: Hetzner CPX 31
 - HORNET 0.4.0 RC12
 - The affected peer used the windows version

**Additional context**
Logs attached from my node and from the peer that got removed on my side:
[log_node.txt](https://github.com/gohornet/hornet/files/4719232/log_node.txt)
[log_peer.txt](https://github.com/gohornet/hornet/files/4719233/log_peer.txt)

**Comment by TeeVeeEss** [comment]
I have a static peer that is shown twice in the output of ``getNeighbors``, once as autopeer and once as static peer:
```
{
      "address": "ip:15621",
      "port": 15621,
      "domain": "ip",
      "numberOfAllTransactions": 12874,
      "numberOfNewTransactions": 1192,
      "numberOfKnownTransactions": 0,
      "numberOfInvalidTransactions": 0,
      "numberOfInvalidRequests": 0,
      "numberOfStaleTransactions": 0,
      "numberOfReceivedTransactionReq": 0,
      "numberOfReceivedMilestoneReq": 1,
      "numberOfReceivedHeartbeats": 1,
      "numberOfSentTransactions": 1134,
      "numberOfSentTransactionsReq": 0,
      "numberOfSentMilestoneReq": 1,
      "numberOfSentHeartbeats": 20,
      "numberOfDroppedSentPackets": 0,
      "connectionType": "tcp",
      "connected": true,
      "autopeeringId": "fa1e2cff3f141e4a"
    },
 {
      "address": "ip:15600",
      "domain": "ip",
      "alias": "discordname",
      "numberOfAllTransactions": 0,
      "numberOfNewTransactions": 0,
      "numberOfKnownTransactions": 0,
      "numberOfInvalidTransactions": 0,
      "numberOfInvalidRequests": 0,
      "numberOfStaleTransactions": 0,
      "numberOfReceivedTransactionReq": 0,
      "numberOfReceivedMilestoneReq": 0,
      "numberOfReceivedHeartbeats": 0,
      "numberOfSentTransactions": 0,
      "numberOfSentTransactionsReq": 0,
      "numberOfSentMilestoneReq": 0,
      "numberOfSentHeartbeats": 0,
      "numberOfDroppedSentPackets": 0,
      "connectionType": "tcp",
      "connected": false
    }
```
And yes, the value of ``port`` is not shown for unconnected peers...

**Comment by WernerderChamp** [comment]
Issue is still present on RC13, happened again during yesterdays spam event (same peer):
[peer loose bug on rc13.txt](https://github.com/gohornet/hornet/files/4719529/peer.loose.bug.on.rc13.txt)

**Comment by muXxer** [comment]
Fixed in #476
